File: java/app/src/main/java/com/ffo/mvp_java/mvp/contract/DemoContract.java
package com.ffo.mvp_java.mvp.contract;

/**
 * @author: huchunhua
 * @create_time: 2018/5/2
 * @change_time:
 * @package: com.ffo.mvp_java.mvp.contract
 * @project: java
 * @mail: huchunhua5@wasu.com
 * @describe: 一句话描述
 */
public interface DemoContract {
    interface View{
        void demoView(String test);
    }

    interface Presenter{
        void demoPresenter();
    }

    interface Model{
        void demoModel(ModelListener modelListener);

        interface ModelListener{
            void completed(String test);
        }
    }
}

File: kotlin/app/src/main/java/pffo/com/mvp/contract/DemoContract.kt
package com.ffo.ipet.mvp.contract

/**
 * @author: huchunhua
 * @create_time: 2018/4/26
 * @change_time:
 * @package: com.ffo.ipet.mvp.contract
 * @project: pet
 * @mail: huchunhua5@wasu.com
 * @describe: 一句话描述
 */
public interface DemoContract{

    /**
     * View
     */
    public interface DemoView{
        fun operationUI(data: Any)
    }

    /**
     * Presenter
     */
    interface DemoPresenter{
        fun loadUI()
    }

    /**
     * Model
     */
    interface IModel{
        fun operationData(operationDataListener : OperationDataListener);

        interface OperationDataListener{
            fun completed(data:Any);
        }
    }
}